block/namespace.go

```diff
@@ -32,6 +33,24 @@ type QualifiedPrefix struct {
 	Prefix           string
 }
 
+func (qk QualifiedKey) Format() string {
+	scheme := ""
+	switch qk.StorageType {
+	case StorageTypeMem:
```

Comment:
Probably out of scope for this PR, but ideally each block adapter should define its oun URI formatter that this function would then use (or better, a formatter that accepts a QualifiedKey and formats it). This way, adding a new block adapter type requires also updating the `QualifiedKey` struct methods - a weird coupling to make.

Reply:
#1164